things/policies/service.go

```diff
@@ -109,26 +109,14 @@ func (svc service) AddPolicy(ctx context.Context, token string, p Policy) (Polic
 	if err := p.Validate(); err != nil {
 		return Policy{}, err
 	}
-	pm := Page{Subject: p.Subject, Object: p.Object, Offset: 0, Limit: 1}
-	page, err := svc.policies.Retrieve(ctx, pm)
-	if err != nil {
-		return Policy{}, err
-	}
-
-	// If the policy already exists, replace the actions
-	if len(page.Policies) == 1 {
-		if err := svc.checkPolicy(ctx, userID, p); err != nil {
-			return Policy{}, err
-		}
-
-		p.UpdatedAt = time.Now()
-		p.UpdatedBy = userID
-		return svc.policies.Update(ctx, p)
-	}
 
 	p.OwnerID = userID
 	p.CreatedAt = time.Now()
 
+	// incase the policy exists, use these for update.
+	p.UpdatedAt = time.Now()
+	p.UpdatedBy = userID
+
```

Comment:
I don't think so. I added it because the full query for an update operation if any is to also update the `updated_at` and `updated_by` field. It we don't have them it will be empty yet we have done an update operation.

Reply:
In fact, @drasko is right. Since this is effectively a policy update, **we do need to invalidate the cache before to prevent incorrect actions**.